electrumsv/network_support/general_api.py

```diff
@@ -273,3 +279,43 @@ async def request_binary_merkle_proof_async(network: Optional[Network], account:
 
     return tsc_proof
 
+
+async def request_transaction_data_async(network: Optional[Network], account: AbstractAccount,
+        tx_hash: bytes) -> Optional[bytes]:
+    """Selects a suitable server and requests the raw transaction.
+
+    Raises `ServerConnectionError` if the remote server is not online (and other networking
+        problems).
+    Raises `GeneralAPIError` if a connection was established but the request errored.
+    """
+    assert network is not None
+    base_server_url = pick_server_for_account(network, account,
+        ServerCapability.TRANSACTION_REQUEST)
+    server_url = f"{base_server_url}api/v1/transaction/"
+    headers = {
+        'Accept':           'application/octet-stream',
+        'User-Agent':       'ElectrumSV'
+    }
+    url = server_url if server_url.endswith("/") else server_url + "/"
+    url += hash_to_hex_str(tx_hash)
+
+    session = await network.get_aiohttp_session()
+    try:
+        async with session.get(url, headers=headers) as response:
+            if response.status == 404:
+                logger.error(f"Transaction for hash {hash_to_hex_str(tx_hash)} "
+                    f"not found")
+                return None
```

Comment:
Raise an exception so this either raises or returns a value consistently, change callers to not look for `None` return value.